src/include/miopen/generic_search.hpp

```diff
@@ -353,128 +353,134 @@ auto GenericSearch(const Solver s, const Context& context, const AnyInvokeParams
-                    elapsed_time /= 5;
-                    if(elapsed_time < best_time)
+                    catch(...)
                     {
-                        MIOPEN_LOG_I('#' << n_current << '/' << n_failed << '/' << n_runs_total
-                                         << ' '
-                                         << elapsed_time
-                                         << " < "
-                                         << best_time
-                                         << ' '
-                                         << current_config);
-                        best_config = current_config;
-                        best_time   = elapsed_time;
-                        n_best      = n_current;
+                        ret = 1;
                     }
-                    else
+
+                    if(ret == 0)
                     {
-                        MIOPEN_LOG_I2(
-                            "Average is not better: " << elapsed_time << " >= " << best_time);
+                        is_passed = true;
+                        elapsed_time /= 5;
+                        if(elapsed_time < best_time)
+                        {
+                            MIOPEN_LOG_I('#' << n_current << '/' << n_failed << '/' << n_runs_total
+                                            << ' '
+                                            << elapsed_time
+                                            << " < "
+                                            << best_time
+                                            << ' '
+                                            << current_config);
+                            best_config = current_config;
+                            best_time   = elapsed_time;
+                            n_best      = n_current;
+                        }
+                        else
+                        {
+                            MIOPEN_LOG_I2(
+                                "Average is not better: " << elapsed_time << " >= " << best_time);
+                        }
                     }
                 }
             }
-        }
 
-        if(ret != 0)
-        {
-            MIOPEN_LOG_E('#' << n_current << " (" << n_runs_total << ") "
-                             << " Failed rc="
-                             << ret);
-            ++n_failed;
+            if(ret != 0)
+            {
+                MIOPEN_LOG_E('#' << n_current << " (" << n_runs_total << ") "
+                                 << " Failed rc="
+                                 << ret);
+                ++n_failed;
+            }
+            heartbeat.Monitor(
+                ret != 0, elapsed_time, n_current, best_time, n_failed, n_runs_total, current_config);
+            ++n_current;
         }
-        heartbeat.Monitor(
-            ret != 0, elapsed_time, n_current, best_time, n_failed, n_runs_total, current_config);
-        ++n_current;
+    }
+    else
+    {
+        MIOPEN_THROW("Search skipped");
```

Comment:
[Recommendation] Explain the reason a bit.
```suggestion
        MIOPEN_THROW("Running kernels on GPU is disabled. Search skipped");
```